sync/fromDisk/questions.js

```diff
@@ -56,11 +56,15 @@ module.exports = {
                             external_grading_files = opts.serverFilesCourse;
                         }
                     }
+                    if ((q.type == 'v3') && (q.partialCredit != null)) {
+                        return callback(new Error(`Question ${qid} has type "v3" and so cannot specify "partialCredit" (it automatically uses partial credit)`));
+                    }
                     var params = {
                         uuid: q.uuid,
                         qid: qid,
                         type: (q.type == 'v3') ? 'Freeform' : q.type,
                         title: q.title,
+                        partial_credit: (q.type == 'v3') ? true : !!q.partialCredit,
```

Comment:
Did you consider making v1/v2 question types the special case (instead of making `v3` the special case), so that we won't have to change this code in future if and when new question types are added?